src/wwwroot/template-editor/adapterService.js

```diff
@@ -0,0 +1,356 @@
+        };
+        return $q.resolve(response);
+      }
+
+      /**
+       * Return a preview image from a video
+       * @method service.getPreviewImage
+       * @argument {object} params - Configuration object
+       * @property {string} params.url - The video url
+       * @property {bool} params.addPlayButton - Flag to check if thumbnail video is added
+       * @property {integer} params.idCampaign - The campaign identifier
+       * @property {integer} params.idTemplate - The template identifier
+       * @property {string} [ThumbnailUrlWithPlay] - The thumbnail url
+       * @example
+       * {
+       *   url: 'http://test.url',
+       *   addPlayButton: true,
+       *   idCampaign: 0,
+       *   idTemplate: 0,
+       *   ThumbnailUrlWithPlay: 'http://vimeocdn.url'
+       * }
+       */
+      function getPreviewImage(params) {
+        console.log("getPreviewImage");
+        var response = {
+              ThumbnailUrl: 'https://i.vimeocdn.com/video/699535210.webp',
+              Title: 'Test video',
+              ProviderName: 'Youtube',
+              ThumbnailUrlWithPlay: 'https://i.vimeocdn.com/filter/overlay?src=' + 'https://i.vimeocdn.com/video/699535210.webp' + '&src=https://integrationstore-b0c3f53658fe7a75.microservice.createsend.com/files/9392B9D9-F380-42FC-9571-7E109B7A1C26/youtube-play-button-overlay.png'
+            };
+        return $q.resolve(response);
+      }
+
+      /**
+       * Return a preview images to image gallery
+       * @method service.getImagesForCampaign
+       * @argument {integer} start - The initial position index 
+       * @argument {integer} howMany - The offset position
+       * @argument {object} query - The query model to filter from gallery
+       * @example
+       * {
+       *   query: ''
+       * }
+       * @argument {string} sortingCriteria - The image sorting criteria
+       */
+      function getImagesForCampaign(start, howMany, query, sortingCriteria) {
+        console.log("getImagesForCampaign");
+        var images = [{
+          name: 'fakeImage1.jpg',
+          lastModifiedDate: '2015-11-14T06:35:42.239Z',
+          size: 1024,
+          type: '.jpg',
+          url: 'http://cdn.fromdoppler.com/mseditor/v1.0.0-build306/uploads/fakeImage1.jpg',
+          thumbnailUrl: 'http://cdn.fromdoppler.com/mseditor/v1.0.0-build306/uploads/fakeImage1.jpg',
+          thumbnailUrl150: 'http://cdn.fromdoppler.com/mseditor/v1.0.0-build306/uploads/fakeImage1.jpg'
+        }];
+        var response = {
+          data:{
+            images: images,
+            count: 1
+          }
+        };
+        return $q.resolve(response);
+      }
+
+      /**
+       * Return an array of promises to delete individual images
+       * @method service.deleteImages
+       * @argument {array} params - Array of image json objects
+       * @property {string} params[0].name - The file name and extension
```

Comment:
Based on [these examples](http://usejsdoc.org/tags-param.html#parameters-with-properties), I think that it should be something like:

```js
/**
 * Return an array of promises to delete individual images
 * @method service.deleteImages
 * @argument {Object[]} params - Array of image json objects
              ˆˆˆˆˆˆˆˆ
 * @property {string} params[].name - The file name and extension
                            ˆˆ
```

Am I missing something?

Reply:
You can see a confirmation here: http://usejsdoc.org/tags-type.html